### PR TITLE
AwesomiumSharp remodeling, additions and improvements.

### PR DESCRIPTION
**Pull request by Perikles**
Major changes in WebView and WebCore. New WPF control (WebControl) introduced that takes advantage of WPF features.

New WPF WebControl Details:

```
 - Instead of wrapping the wrapper (meaning, wrapping WebView), we
   become the initial wrapper ourselves and directly implement all
   C API just as WebView does. This improves performance, allows more
   control and allows the developer to access the full API directly
   from this control.

 - Just like in the other files edited, here too, many changes
   with respect to standard .NET guidelines and naming convension
   were made.

 - Properties are available as dependency properties. This allows
   the developer to easily bind to properties and monitor their
   status in triggers etc.

 - WebControl is a now a FrameworkElement as it should be. This means
   styling is possible but templating is not. No useless elements in
   the visual tree and no children such as images. We render directly
   on the element's surface and override our own events' triggers.

 - WebControl includes an internal auto updater just as the one added
   in WebCore. It monitors WebCore's auto updater and starts or stops
   as needed.

 - Part of the Find logic is now handled by this class making it
   more straightforward. FindNext is added.

 - Extensive pro-exception verification of validity before every API
   call was added here too.

 - Most of the control's methods can be called using routed commands.
   We bind to many of the already available Application and Navigation
   commands, and we cover more functionality by providing our own
   commands through WebControlCommands. This makes it easy for the UI
   developer to manipulate the control directly from XAML. See
   InitializeCommandBindings() to get an idea of what we support.
```

WebCore Changes Details:

```
- Many changes with respect to standard .NET guidelines and naming
  convention were made. These include, among others:
      * Get/Set accessors were turned into Properties wherever
        this was appropriate.
      * The names of many members were changed to follow proper
        naming convention.
      * WebCoreConfig is created, documented and taken out of
        the class.

- Background auto-updating is added and can be configured during
  initialization through WebCoreConfig, or afterwards through
  the AutoUpdate property.

- Extensive pro-exception verification of validity before every API
  call is added.
```

WebView Changes Details:

```
- Many changes with respect to standard .NET guidelines and naming
  convention were made. These include, among others:
      * Get/Set accessors were turned into Properties wherever
        this was appropriate.
      * The names of many events were changed to follow proper
        naming convention.
      * Protected event triggers were added.
      * Former inner classes (such as the various EventArgs) are
        taken outside the class and moved to separate files. They
        can be found in the EventArgs project folder. All these
        classes have also been edited following the same guidelines.

- A base modeling class (ViewModel) implementing INotifyPropertyChanged
  is added and subclassed. This makes WebView MVVM friendly.

- Part of the Find logic is now handled by this class making it
  more straightforward. FindNext is added.

- Extensive pro-exception verification of validity before every API
  call is added.
```

Note to the original authors:

```
- Please contact me @: perikles@stephanidis.net to let me know
  what you think of my changes.
  If you do not wish to pull them, please let me know. 
  I will publish a separate project here or in CodePlex.
- I would love to contribute to the AwesomiumSharp wrapper
  officially. I'm an experienced .NET and WPF developer and teacher.
```

Thank You!
Perikles C. Stephanidis

### CODE REVIEW

File: VBWPFSample/MainWindow.xaml [hunk 1]
```diff
@@ -0,0 +1,171 @@
```
Comment (Perikles) [inline VBWPFSample/MainWindow.xaml:0]: This file (and the whole VBWPFSample), demonstrate the use of the new WPF WebControl.
Please read the README file included in this sample's folder.

**Comment by lanwin** [comment]
+1

**Comment by adamkhrona** [comment]
Wow, there are some really awesome contributions! Great work, reviewing this now.

**Comment by Perikles** [comment]
Thank you for accepting the Pull Request and for your comments. I will keep working on this and committing changes. I have sent you a message asking for some info regarding future commits.

**Pull request merged**